### DISCUSSION TITLE
Trying to understand the parabola graph

### DISCUSSION THREAD

**Discussion by Eneswar** [DylanHojnoski/obsidian-graphs#22] [General]
Hello, so im trying to make a simple y=x^2 graph, like so:

![image](https://github.com/DylanHojnoski/obsidian-graphs/assets/22943781/73642c77-c36a-462d-9ece-1c8148052b5e)

The problem Im having is making it narrower, and also changing some of the numbers makes no difference.

![image](https://github.com/DylanHojnoski/obsidian-graphs/assets/22943781/3aa3321c-ce99-48ad-bd97-0b88397da298)

Basically its too wide, it should be when x is 2 y should be 4, and I have been fiddling with the numbers for 20 minutes now without any success, so im just wondering is this possible to do?

```graph
bounds: [-5, 10, 5, -5]
drag: false
showNavigation: false
keepAspectRatio: true
elements: [
	{type: "parabola", def: [[0,1], [[1,-1],[0,-1]]]},
]
```

Would be really amazing if I could use these `y=x^(2)-2x+1` and it would show me the graph but I assume thats not possibe :D

**Comment by DylanHojnoski** [comment]
You can use the `functiongraph` type which allows you to just put in a function using a string like `"f:x**2-2*x+1"`. Anything after `f:` gets evaluated in the function and `**` is used for exponents.


```
bounds: [-5, 10, 5, -5]
drag: false
showNavigation: false
keepAspectRatio: true
elements: [
	{type: "functiongraph", def: ["f:x**2-2*x+1"]},
]
```

**Reply by Eneswar** [reply]
I had no idea you could do that, that is amazing, thank you so much!

**Comment by Eneswar** [comment]
How can I name a point? So its not just "A" ?

**Reply by DylanHojnoski** [reply]
You can set the `name` attribute like `att: {name: "Name"}` and that will change the text that is displayed next to a point.

**Comment by Eneswar** [comment]
Can I like buy you a coffee or something? I dont see a link on the github page.

**Reply by DylanHojnoski** [reply]
I just activated [GitHub Sponsors]( https://github.com/sponsors/DylanHojnoski) you can donate there if you like.

**Comment by Eneswar** [comment]
What about a circular function? Example (x-3)^{2}+(y+2)^{2}=3^{2}

```
bounds: [-10, 10, 10, -10]
drag: false
showNavigation: false
keepAspectRatio: true
elements: [
	{type: "functiongraph", def: ["f:(x-3)**2+(y+2)**2=3**2"]},
]
```

Im getting SyntexError: Invalid left-hand side in assignment

**Reply by DylanHojnoski** [reply]
You can't use `=` in the function. There is however the `circle` type where you give the point and then the radius.
```
{type: "circle", def: [[3,-2], 2]}
```

Or you can use the `implicitcurve` where you can use both `x` and `y`  in the equation.
```
{type: "implicitcurve", def: ["f:(x-3)**2 + (y+2)**2 - 2"]}
```

**Comment by Eneswar** [comment]
Didnt want to start a new thread... Just a question, is it possible to create a dotted line of some sort?

**Reply by DylanHojnoski** [reply]
You can use  `att: { dash: 1 }`  to make any line a dotted line.

**Reply by Eneswar** [reply]
Amazing, thank you very much. Im sorry for asking so many question, I just love this plugin.
If you dont mind I have one more :D

![image](https://github.com/user-attachments/assets/5dedbf30-a94a-4ca8-aeb0-b8dca6a725b3)

Is there a way to show the x-axis as radians? so pi and pi/2 etc?

**Reply by DylanHojnoski** [reply]
No problem for asking questions I'm glad you're enjoying the plugin. You can change the scale of the axis and the symbol the code below does what you want. Right now you can't set a function for the scale so you can't set it directly as PI but I will look into changing that and an estimation will probably be close enough.

````
```graph
bounds: [-5, 5, 5, -5]
defaultAxes:
  x:
    ticks:
      scale: 3.1459
      scaleSymbol: "π"
      label:
        toFraction: true
elements: [
	{type: "functiongraph", def: ["f:sin(x)"]}
]
```
````

**Comment by Eneswar** [comment]
Hello and sorry I have another question lol.

Im trying to name an arrow element, preferable where the arrow head is but it doesnt matter where exactly.

`{type: "arrow", def: [[-2,-2], [4,4]]},`

I tried the usual `att: {name: "u"}` but nothing shows up?

**Reply by DylanHojnoski** [reply]
Set the `withLabel: true` attribute and then it should be visible.
